plan/match_property.go

```diff
@@ -231,10 +231,7 @@ func (p *Selection) matchProperty(prop *requiredProperty, childPlanInfo ...*phys
 		res.count = uint64(float64(res.count) * selectionFactor)
 		return res
 	}
-	np := *p
-	np.SetChildren(childPlanInfo[0].p)
-	count := uint64(float64(childPlanInfo[0].count) * selectionFactor)
-	return &physicalPlanInfo{p: &np, cost: childPlanInfo[0].cost, count: count}
+	return childPlanInfo[0]
```

Comment:
Here the `childPlanInfo` is actually not child, it is the selection itself? 
Then the `matchProperty` is not used as it is supposed to be.